src/status_im2/contexts/communities/discover/view.cljs

```diff
@@ -241,11 +228,21 @@
         featured-communities
         @view-type]])))
 
+(defn f-discover
+  []
+  (fn []
+    (rn/use-effect
+     (fn []
+       (rf/dispatch [:fetch-contract-communities]))
+     [])
+    (let [featured-communities (rf/sub
+                                [:communities/featured-contract-communities])]
+      [rn/view
+       {:style (style/discover-screen-container (colors/theme-colors
+                                                 colors/white
+                                                 colors/neutral-95))}
+       [discover-screen-content featured-communities]])))
+
 (defn discover
   []
-  (let [featured-communities (rf/sub [:communities/featured-contract-communities])]
-    [rn/view
-     {:style (style/discover-screen-container (colors/theme-colors
-                                               colors/white
-                                               colors/neutral-95))}
-     [discover-screen-content featured-communities]]))
+  [:f> f-discover])
```

Comment:
@cammellos @J-Son89 
My opinion on this:

- `f-discover` is a form-2 component, but it's not required if we are already using use-effect.
- the `use-effect` call will be only executed on mount and unmount, but we are not performing cleaning, so we only care about mount and I think it can be avoided.
- So that it's not required to be a functional component.

So I'd suggest doing this:
```clojure
(defn discover
  []
  (let [_ (rf/dispatch [:fetch-contract-communities])] ; Executed only once when the component is mounted
    (fn []
      (let [featured-communities (rf/sub [:communities/featured-contract-communities])]
        [rn/view
         {:style (style/discover-screen-container (colors/theme-colors
                                                   colors/white
                                                   colors/neutral-95))}
         [discover-screen-content featured-communities]]))))
```

(it'd be almost the same as using `reagent/with-let`)

Also, that `let` is not required, the component could look as follows:
```clojure
(defn discover
  []
  (rf/dispatch [:fetch-contract-communities]) ; The same as before, but without let
  (fn []
    (let [featured-communities (rf/sub [:communities/featured-contract-communities])]
      [rn/view
       {:style (style/discover-screen-container (colors/theme-colors
                                                 colors/white
                                                 colors/neutral-95))}
       [discover-screen-content featured-communities]])))
```

But this might break the form-2 mental model for other devs :thinking:, we could add an unused binding name (a var name starting with `_`), our linter is aware of this, so we won't get any warning:

```clojure
;; Ignored let binding:
(let [_fetch-communities (rf/dispatch [:fetch-contract-communities])]
  (fn [] ,,,))

```

Feel free to take any of those, I just think the current code looks a little more complex than needed :sweat_smile:

Reply:
no strong opinion on my side, I'll go with the team convention, thanks!